### PR TITLE
web: migrate to web platform- packages

### PR DESCRIPTION
**Pull request by BRKalow**
Migrate away from @hashicorp/nextjs-scripts to the new suite of @hashicorp/platform-* packages

[_Created by Sourcegraph campaign `bkalow/migrate-nextjs-scripts-packages`._](https://sourcegraph.hashi-mktg.com/users/bkalow/campaigns/migrate-nextjs-scripts-packages)

### CODE REVIEW

**Comment by izaaklauer** [comment]
Hey one questions about this:

We're using next-js explicitly in our make file to format our mdx files in CI (https://github.com/hashicorp/waypoint/blob/main/Makefile#L137). Do you know offhand what the change would be there to migrate to the new packages?

**Comment by BRKalow** [comment]
Good catch! `website-mdx` is fixed up now after my latest commit, so no changes should be necessary for that workflow 😄

File: website/pages/downloads/index.jsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+import ProductDownloadsPage from '@hashicorp/react-product-downloads-page'
+import { generateStaticProps } from '@hashicorp/react-product-downloads-page/server'
 import styles from './style.module.css'
 
 export default function DownloadsPage({ releases }) {
   return (
     <span className={styles.downloads}>
       <HashiHead title={`Downloads | ${productName} by HashiCorp`} />
-      <ProductDownloader
+      <ProductDownloadsPage
```
Comment (jescalan) [inline website/pages/downloads/index.jsx:12]: wrapping span and head can be dropped here since this is now a full page component

**Review by jescalan — CHANGES_REQUESTED** [review]
just one small change with the downloads page

File: website/pages/downloads/index.jsx (hunk 1 above)
Comment (BRKalow) [reply website/pages/downloads/index.jsx:12]: Aha I knew I migrated another usage somewhere...thank you 🙏

**Review by jescalan — APPROVED** [review]

**Review by izaaklauer — APPROVED** [review]

**Comment by izaaklauer** [comment]
Oh actually, when I re-run `npm install` from the website directory after this change, I get a large change to `package-lock.json`. Is this expected?

**Comment by BRKalow** [comment]
@izaaklauer What's your `npm -v`? If it's `7`, it introduces a new lockfile format which causes quite a few changes. More info if you're curious: https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json#lockfileversion

It should be fine all things considered, we've just gotta start enforcing npm v7.

**Comment by BRKalow** [comment]
@zchsh Let me know if I can help with the updated global styles rollout here. I think it should fix the terminal background color issue 👀

**Comment by izaaklauer** [comment]
> @izaaklauer What's your `npm -v`

@BRKalow Ah yup, it's `7.19.1`, and I can see that the lockFileVersion bumped from 1 to 2. Might make sense for us to commit those changes to embrace the new npm version.

**Pull request merged**